### PR TITLE
Update contributor ladder

### PR DESCRIPTION
**Pull request by davidmirror-ops**
In this PR, multiple changes are proposed:

1. Improve the definition of requirements and responsibilities for each role
2. Introduce the Collaborator role, lowering the barrier between the general community and the Contributor privileges
3. Expand the current status of the path to Maintainership

### CODE REVIEW

**Review by pingsutw — APPROVED** [review]

**Review by cosmicBboy — APPROVED** [review]

File: GOVERNANCE.md [hunk 1]
```diff
@@ -37,42 +37,71 @@ The following table lists the roles we use within the Flyte community. The table
...
+        <p>Active on Slack posting/answering questions</p>
+        <p>Promote the project creating content/delivering talks/demos</p>       
+             
+   </td>
+    <td>
+        <p>Can submit PR reviews</p>
+        <p>May close/open/reassign Issues</p>
+        <p>May request PR reviews</p>
+        <p>Can mark duplicate Issues/PRs</p>
+        <p>Can create forks</p>
```
Comment (fg91) [inline GOVERNANCE.md:64]: Shouldn't everyone be allowed to create forks? How do you get to a first merged contribution without having created a fork?

File: GOVERNANCE.md [hunk 2]
```diff
@@ -121,8 +150,20 @@ The following table lists the roles we use within the Flyte community. The table
 ### **Community participant**
 You do not need to be a member of the Flyte team to contribute. Anyone can help by using and talking about Flyte, participating in discussions, answering questions on Slack/Stack Overflow, opening issues, submitting PRs, etc. All of these are fantastic and welcome contributions to the project.
 
+### **Collaborator**
+Collaborators are individuals who demonstrate an active interest on the project and are willing to help other community members.
```
Comment (fg91) [inline GOVERNANCE.md:154]:
```suggestion
Collaborators are individuals who demonstrate an active interest in the project and are willing to help other community members.
```

**Review by fg91 — CHANGES_REQUESTED** [review]
Shouldn't also community members be able to create forks? How could they get their first merged contributions otherwise?

Apart from that LGTM

File: GOVERNANCE.md (hunk 1 above)
Comment (wild-endeavor) [reply GOVERNANCE.md:64]: yeah you can always fork, that's a github thing.  i think this should just say that they should create a fork to create first prs.

File: GOVERNANCE.md (hunk 2 above)
Comment (davidmirror-ops) [reply GOVERNANCE.md]: Thanks!

File: GOVERNANCE.md (hunk 1 above)
Comment (davidmirror-ops) [reply GOVERNANCE.md]: Right, made it a "baseline" privilege for everyone, which is really a GH feature.

**Review by bstadlbauer — APPROVED** [review]
Looks good to me, thank you!

**Pull request merged**